Render oneway markers inline in dashed lines

tyrasd opened the pull request:
This fixes the awkward overlaps of oneway arrows in dashed lines with inconsistent spacing, and hard to choose well contrasting colors for the arrows (see #9143). Now, for lines with dashed rendering style (e.g. railway lines, some footways/paths, runways) the oneway arrows are rendered in place of one of the line's dashes. This allows the oneway arrows to be rendered in the same color as the line itself, leading to a more elegant, less noisy rendering for these lines. 

Below are rendering examples of various lines with different combinations of `oneway` tags (_none_, `yes`, `-1`, `alternating`), from left to right: regular roads, railways, railway bridges, pedestrian bridge, pedestrian road, runways, road under construction, footpaths/cycleway, steps and ladders, unmarked pedestrian/cycleway crossing.

after:
![image](https://github.com/user-attachments/assets/0512f3ff-d003-4da0-951d-769dc2b7eda8)

before:
![image](https://github.com/user-attachments/assets/5df3d532-4329-48b9-ab9e-c0cb9f190111)


(this PR replaces #10846)

tyrasd committed:
render oneway markers as inline dashes for dashed strokes

requires slight tweaking of colors and dash lengths and dash spacings

tyrasd committed:
add type definitions for projection object, modernize code slightly

k-yle commented on modules/svg/defs.js:
```diff
@@ -27,13 +27,13 @@ export function svgDefs(context) {
                 .attr('id', `ideditor-oneway-marker-${name}`)
                 .attr('viewBox', '0 0 10 5')
-                .attr('refX', 2.5)
+                .attr('refX', 4)
```
this is a much nicer solution :) my only suggestion would be to keep the original spacing between bidirectional arrow heads, since it's a bit cramped now:

original:
<img width="353" alt="image" src="https://github.com/user-attachments/assets/434848be-0d45-4313-b302-46c540f10dd0" />

this PR:
<img width="353" alt="image" src="https://github.com/user-attachments/assets/8aa260dc-7b0c-4320-8c9f-0fad85197a31" />

tyrasd replied:
Yeah, the new version does look a bit squished… I needed to shrink the length of the double-sided arrows a bit though, in order to fit them into the existing dash spacings of the different line styles.

I experimented a bit with the shape of the arrows, and found two that seem to work reasonably well even in the more condensed form for the bidirectional arrows:

1. original [`M 5,3 L 0,3 L 0,2 L 5,2 L 5,0 L 10,2.5 L 5,5 z`](https://svg-path-visualizer.netlify.app/#M%205%2C3%20L%200%2C3%20L%200%2C2%20L%205%2C2%20L%205%2C0%20L%2010%2C2.5%20L%205%2C5%20z) (with `refX=2.5`)
2. first version of this PR: same shape as above, but `refX=4`
3. arrow alternative 1 [`M 6,3 L 0,3 L 0,2 L 6,2 L 5,0 L 10,2.5 L 5,5 z`](https://svg-path-visualizer.netlify.app/#M%206%2C3%20L%200%2C3%20L%200%2C2%20L%206%2C2%20L%205%2C0%20L%2010%2C2.5%20L%205%2C5%20z) (`refX=4`)
4. arrow alternative 2 [`M 6.5,3 L 0,3 L 0,2 L 6.5,2 L 5.5,0 L 10,2.5 L 5.5,5 z`](https://svg-path-visualizer.netlify.app/#M%206.5%2C3%20L%200%2C3%20L%200%2C2%20L%206.5%2C2%20L%205.5%2C0%20L%2010%2C2.5%20L%205.5%2C5%20z) (`refX=4`)

![image](https://github.com/user-attachments/assets/9fc8365f-732e-4811-9fa5-cd2907dee746)

Here's the test suite from above again with alternatives 1 and 2:

![image](https://github.com/user-attachments/assets/41ffd292-8a55-4e96-a4c8-f30c473f1701)

![image](https://github.com/user-attachments/assets/2d8d72dd-1817-4a83-a0c5-a8ff9d27b5de)

PS: I also bumped up the opacity to 100% for all arrows in the last two screenshots, as I think the slight transparency of the arrows is not necessary and only caused a minor issue where the two arrows overlap in for bidirectional lines. But maybe it would be worth toning down the color for the `black` arrows to something like `#333` to tone down the contrast a little bit, and bring it back in line with how the arrows previously appeared on typical lines.

tyrasd committed:
also use padded clip extent for oneway markers

otherwise they would be rendered out of sync of the line's dashes when the feature extends outside of the viewport

tyrasd committed:
tweak oneway arrow shape and colour slightly

see https://github.com/openstreetmap/iD/pull/10849#discussion_r1986294976 (this is "arrow alternative 1")

tyrasd committed:
Merge branch 'develop' into onewaymarkers-dashed-inline

tyrasd committed:
add changelog
